torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -242,6 +244,72 @@ def aot_dispatch_base(
     return compiled_fn
 
 
+def collect_donated_buffer_metadata_from_tensor(
+    fw_inputs: List[torch.Tensor],
+    fw_outputs: List[torch.Tensor],
+    bw_outputs: List[torch.Tensor],
+    fw_saved_tensor_range: Tuple[int, int],
```

Comment:
and maybe just to make this function a bit easier to reason about: pass `user_fw_outs` and `saved_activations` as two separate lists? (the caller is in charge of the annoying indexing)